pkg/image/api/helper_test.go

```diff
@@ -96,14 +107,36 @@ func TestParseDockerImageReference(t *testing.T) {
 			ID:        "sha256:3c87c572822935df60f0f5d3665bd376841a7fcfeb806b5f212de6a00e9a7b25",
 		},
 		{
-			From:      "myregistry.io/foo",
-			Registry:  "myregistry.io",
+			From:     "myregistry.io/foo",
+			Registry: "myregistry.io",
+			Name:     "foo",
+		},
+		{
+			From:     "localhost/bar",
+			Registry: "localhost",
+			Name:     "bar",
+		},
+		{
+			From:      "docker.io/library/myapp",
+			Registry:  "docker.io",
+			Namespace: "library",
+			Name:      "myapp",
+		},
+		{
+			From:      "docker.io/myapp",
+			Registry:  "docker.io",
 			Namespace: DockerDefaultNamespace,
-			Name:      "foo",
+			Name:      "myapp",
 		},
 		{
-			From:      "localhost/bar",
```

Comment:
Why did the local host test get removed. I really want to touch as few existing tests as possible here